bionlp/preprocess/parser.py

```diff
@@ -0,0 +1,96 @@
+from abc import ABC, abstractmethod
+import json
+import os
+
+from bionlp.tags import IOB2_Tag
+
+class AbstractDatasetParser(ABC):
+    @abstractmethod
+    def parse_datasets(self, input_dir, output_dir):
+        raise NotImplementedError
+
+    @abstractmethod
+    def parse_file(self, input_file, output_file):
+        raise NotImplementedError
+
+
+class BratParser(AbstractDatasetParser):
+    # this is implemented in the BratParser class instead of the abstract parent class because,
+    # for Brat, we only want to regard the input files with file extension .ann
+    def parse_datasets(self, input_dir, output_dir):
+        # this methodology only expects files in the top level of the given input directory, not subdirs
+        _, _, input_files = next(os.walk(input_dir))
+        for input_file in input_files:
+            if input_file.endswith(".ann"):
+                input_file_abs_path = os.path.join(input_dir, input_file)
+                output_file_abs_path = os.path.join(output_dir, os.path.splitext(input_file)[0] + ".json")
+                self.parse_file(input_file_abs_path, output_file_abs_path)
+
+    def parse_file(self, input_file, output_file):
+        total_tags_list = []
+        with open(input_file, "r", encoding="utf8") as f_in:
+            tag_chain_beginning_index = None
+            tag_chain_end_index = None
+            tag_chain_token_values = ""
+            tag_chain_tag = None
+            tag_chain_id = -1
+            for line in f_in.readlines():
+                cleaned_line = line.rstrip()
+                brat_index, tag_and_indices, token_value = cleaned_line.split("\t")
```

Comment:
I think it's readable to omit the `cleaned_line` variable and declare `... = line.rstrip().split("\t")`